### ISSUE TITLE
DNS queries are not proxied when a sock5 proxy is being used in auto switch mode

### ISSUE THREAD

**Issue by simonxu** [FelisCatus/SwitchyOmega#572]
The setup: Windows 10, Chrome 64bit, SwitchOmega 2.3.15

In auto switch mode, when accessing a site whose domain name matches the sock5 proxy rule, sock5 proxy will be used but dns queries do not use the sock5 proxy.  This is a problem when the domain name can only be resolved through the sock5 proxy.

When SwitchOmega is configured to directly use the sock5 proxy mode, the domain name can be resolved through the proxy.

**Comment by celeron533** [comment]
I have the similar environment but never meet this problem....

Please check whether the DNS cache is poisoned.

![image](https://cloud.githubusercontent.com/assets/3608762/9604103/eb5dff06-50e7-11e5-8cee-67460d6c2e9c.png)
![image](https://cloud.githubusercontent.com/assets/3608762/9604121/15b97c94-50e8-11e5-8613-c201e894a58b.png)

**Comment by fherbert** [comment]
I have managed to replicate this issue of remote dns not being resolved through the socks proxy when using auto switch mode. I get "DNS_PROBE_FINISHED_NXDOMAIN" error if dns cannot be resolved by my local machine. But using firefox and selecting 'Use Remote DNS" this works as expected and resolves the dns through the socks proxy.

**Comment by FelisCatus** [comment]
https://github.com/FelisCatus/SwitchyOmega/wiki/DNS-and-SOCKS-proxy

**Comment by simonxu** [comment]
I have read the page.  How can I solve this issue with Chrome (Google Chrome on Windows)?

**Comment by FelisCatus** [comment]
As shown in the wiki, you can disable DNS prefetching completely by unchecking the following option in your Chromium settings (under "Show advanced settings..."):

```
[ ] Predict network actions to improve page load performance
```

This option should be there for any operating system and most variants of Chromium, including Google Chrome.

**Comment by simonxu** [comment]
Sorry I didn't find the entry in Chrome settings because it was not obvious (they are not in English), now I have found it, thank you for your help!

**Comment by tpmullan** [comment]
That still doesn't work.  Why doesn't the extension configure chrome to proxy dns?  There isn't a simple setting to change in chrome and adding startup options that hard code a proxy isn't a solution.

**Comment by Nexolight** [comment]
`"Predict network actions to improve page load performance"`

This doesn't change anything at all. error: -105 (ERR_NAME_NOT_RESOLVED)
However when the DNS would work trough the proxy this would be fine.

**Comment by HikariWS** [comment]
I looked on that wiki, but it seems that “Use a prediction service to load pages more quickly” was removed from Chromium long ago.

Any chance of having this working? I use SOCKS5 to reach a TOR relay, and .onion domains don't work, because DNS is being fetched from outside the proxy, while only TOR network is able to handle them.

It'd be awesome to have something like a checkbox related to each profile, so that we can choose on each profile if DNS should be fetched from inside or outside of its proxy.
